repository/src/main/resources/db/changelog/db.changelog-rdm-2917.xml

```diff
@@ -0,0 +1,504 @@
+        <!-- END - event_acl table -->
+
+
+
+        <!-- START - state_acl table -->
+        <!-- create the new state_acl table -->
+        <createTable tableName="state_acl">
+            <column name="id" type="serial">
+                <constraints nullable="false"/>
+            </column>
+            <column name="state_id" type="integer">
+                <constraints nullable="false"/>
+            </column>
+            <column name="user_role_id" type="integer">
+                <constraints nullable="true"/>
+            </column>
+            <column name="create" type="boolean">
+                <constraints nullable="false"/>
+            </column>
+            <column name="read" type="boolean">
+                <constraints nullable="false"/>
+            </column>
+            <column name="update" type="boolean">
+                <constraints nullable="false"/>
+            </column>
+            <column name="delete" type="boolean">
+                <constraints nullable="false"/>
+            </column>
+            <column name="live_from" type="date" >
+                <constraints nullable="true"/>
+            </column>
+            <column name="live_to" type="date" >
+                <constraints nullable="true"/>
+            </column>
+            <column name="created_at" type="datetime" defaultValue="now()">
+                <constraints nullable="false"/>
+            </column>
+        </createTable>
+
+        <addPrimaryKey columnNames="id"
+                       constraintName="pk_state_acl"
+                       tableName="state_acl"/>
+
+        <!-- Add foreign key for state_id column linking to state table -->
+        <addForeignKeyConstraint baseColumnNames="state_id"
+                                 baseTableName="state_acl"
+                                 constraintName="fk_state_acl_state_id_state_id"
+                                 referencedColumnNames="id"
+                                 referencedTableName="state"/>
+
+        <!-- Add foreign key for user_role_id column linking to user_role table -->
+        <addForeignKeyConstraint baseColumnNames="user_role_id"
+                                 baseTableName="state_acl"
+                                 constraintName="fk_state_acl_user_role_id_user_role_id"
+                                 referencedColumnNames="id"
+                                 referencedTableName="user_role"/>
+
+        <addUniqueConstraint columnNames="state_id, user_role_id"
+                             constraintName="unique_state_acl_state_id_user_role_id"
+                             tableName="state_acl"/>
+
+        <!-- copy table data to the new state_acl table -->
+        <sql>
+            insert into state_acl select * from state_user_role;
+        </sql>
+
+        <!-- Add new role_id column to the state_acl table -->
+        <addColumn tableName="state_acl">
```

Comment:
I don't see the benefit of splitting the column creation from the table creation